### PR TITLE
feat(tier4_autoware_api_launch): add tier4_autoware_api_launch package

### PR DESCRIPTION
**Pull request by mitsudome-r**
Signed-off-by: mitsudome-r <ryohsuke.mitsudome@tier4.jp>

## Description
This PR adds tier4_autoware_api_launch package to the repository which is required to connect Autoware with TIER IV's scenario simulator. 
For the details, please refer to https://github.com/autowarefoundation/autoware.universe/issues/656.

## Related links
https://github.com/autowarefoundation/autoware.universe/issues/656

## Tests performed
Follow the [installation instruction](https://autowarefoundation.github.io/autoware-documentation/main/installation/autoware/source-installation/) from the documentation and confirm that it build without an error.

## Notes for reviewers
This PR depends on https://github.com/autowarefoundation/autoware/pull/156

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: launch/tier4_autoware_api_launch/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+## Package Dependencies
+
+Please see `<exec_depend>` in `package.xml`.
+
+## Usage
+
+You can include as follows in `*.launch.xml` to use `control.launch.py`.
+
+```xml
+  <include file="$(find-pkg-share autoware_api_launch)/launch/tier4_autoware_api.launch.xml">
```
Comment (kenji-miyake) [inline launch/tier4_autoware_api_launch/README.md:16]:
```suggestion
  <include file="$(find-pkg-share tier4_autoware_api_launch)/launch/autoware_api.launch.xml">
```

**Review by kenji-miyake — COMMENTED** [review]
Related:
- https://github.com/tier4/autoware_launch/tree/tier4/universe/autoware_api_launch
- https://github.com/tier4/tier4_ad_api_adaptor

**Comment by kenji-miyake** [comment]
@mitsudome-r Where did you copy these files from?
It seems to be a bit different from https://github.com/tier4/autoware_launch/tree/tier4/universe/autoware_api_launch. :thinking: 

@isamu-takagi Could you check this just in case? :pray:

**Comment by mitsudome-r** [comment]
> @mitsudome-r Where did you copy these files from? It seems to be a bit different from https://github.com/tier4/autoware_launch/tree/tier4/universe/autoware_api_launch. thinking

I apologize that I forgot to mention in the description. I have copied from the same link, but I removed the ones that are not necessary for the scenario simulator at the moment. I was thinking it might be better if we add:
* path_distance_calculator once we agree on how we handle safety
* rosbridge_websocket_launch once we agree on how we communicate through web interface

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**